core/shared/src/main/scala/scalaz/zio/ZSchedule.scala

```diff
@@ -239,7 +245,7 @@ trait ZSchedule[-R, -A, +B] extends Serializable { self =>
    * using the minimum of the delays of the two schedules.
    */
   final def ||[R1 <: R, A1 <: A, C](that: ZSchedule[R1, A1, C]): ZSchedule[R1, A1, (B, C)] =
-    combineWith(that)(_ || _, _ min _)
+    combineWith(that)(_ || _, (a, b) => Min(a, b))
```

Comment:
I'd add `max` and `min` functions to `Delay` so you don't have to change this code.